Source/WebCore/html/FormListedElement.cpp

```diff
@@ -108,7 +108,8 @@ static RefPtr<HTMLFormElement> findAssociatedForm(const HTMLElement& element, HT
             if (!newFormCandidate)
                 return nullptr;
             if (&element.traverseToRootNode() == &element.treeScope().rootNode()) {
-                ASSERT(&element.traverseToRootNode() == &newFormCandidate->traverseToRootNode());
+                ASSERT((element.document().settings().shadowRootReferenceTargetEnabled() && &element.traverseToRootNode() == &(element.treeScope().retargetToScope(*newFormCandidate))->treeScope().rootNode())
```

Comment:
This assertion is really long and hard to parse. I think we can use if statement here instead:
```cpp
if (element.document().settings().shadowRootReferenceTargetEnabled())
    ASSERT(&element.traverseToRootNode() == &(element.treeScope().retargetToScope(*newFormCandidate))->treeScope().rootNode()));
else
    ASSERT(&element.traverseToRootNode() == &newFormCandidate->traverseToRootNode());
```